docker/docker.ts

```diff
@@ -458,16 +459,14 @@ async function tagAndPushImageAsync(
     // Ensure we have a unique target name for this image, and tag and push to that unique target.
     await doTagAndPushAsync(createTaggedImageName(repositoryUrl, tag, imageId));
 
-    if (tag) {
-        // user provided a tag themselves (like "x/y:dev").  In this case, also tag and push
-        // directly to that 'dev' tag.  This is not going to be a unique location, and future pushes
-        // will overwrite this location.  However, that's ok as there's still the unique target we
-        // generated above.
-        //
-        // We don't need to do this for the main image we tagged and pushed as the repo will
-        // automatically give is a :latest tag that serves the same purpose.  So this tagged/pushed
-        // build will simply act as the 'latest' pointer for this specific tag.
-        await doTagAndPushAsync(`${repositoryUrl}:${tag}`);
+    // If the user provided a tag themselves (like "x/y:dev") then also tag and push directly to
+    // that 'dev' tag.  This is not going to be a unique location, and future pushes will overwrite
+    // this location.  However, that's ok as there's still the unique target we generated above.
+    //
+    // Note: don't need to do this if imageId was 'undefined' as the above line will have already
+    // taken care of things for us.
+    if (tag !== undefined && imageId !== undefined) {
+        await doTagAndPushAsync(createTaggedImageName(repositoryUrl, tag, /*imageId:*/ undefined));
```

Comment:
i added the imageId check here to prevent us from double pushing. i.e. if the call into this function has a tag, but no imageId, then we were executing the same code twice.  once on line 459, and once here.  Now, we don't bother with this tag/push here if we jsut did it above.